### PR TITLE
LPS-179298 [PRM] Fetching data for Partner Level Fragment

### PR DESCRIPTION
**Pull request by JoMaAlves**

### CODE REVIEW

**Comment by JoMaAlves** [comment]
ci:test:sf

**Comment by JoMaAlves** [comment]
ci:test:relevant

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js [hunk 1]
```diff
@@ -9,226 +9,214 @@
...
+
+					if (
+						accountData.partnerLevel !==
+						PartnershipLevels.AUTHORIZED
+					) {
+						if (accountData.solutionDeliveryCertification) {
+							checkedItems[
+								'solutionDeliveryCertification'
+							] = true;
+						}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js:81]: instead we have a if with a true assignment, you can pass the verification directly.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js [hunk 2]
```diff
@@ -9,226 +9,214 @@
...
-				if (data.marketing) {
-					checkedItems['marketing'] = true;
-				}
+						if (
+							accountData.partnerLevel !==
+							PartnershipLevels.SILVER
+						) {
+							if (accountData.marketingPlan) {
+								checkedItems['marketingPlan'] = true;
+							}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js:89]: instead we have a if with a true assignment, you can pass the verification directly.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js [hunk 3]
```diff
@@ -9,226 +9,214 @@
...
+
+							if (accountData.marketingPerformance) {
+								checkedItems['marketingPerformance'] = true;
+							}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js:93]: instead we have a if with a true assignment, you can pass the verification directly.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js [hunk 4]
```diff
@@ -9,226 +9,214 @@
...
+
+							if (
+								accountData.partnerLevel ===
+									PartnershipLevels.PLATINUM &&
+								accountData.growthARR + accountData.renewalARR >
+									0 &&
+								accountData.aRRAmount >=
+									accountData.growthARR +
+										accountData.renewalARR
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js:123]: Pass those math in different variables

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js [hunk 5]
```diff
@@ -9,226 +9,214 @@
...
+										role.name === 'Partner Marketing User'
+								)
+							) {
+								countHeadcount['partnerMarketingUser'] += 1;
+							}
+
+							if (
+								user.accountBriefs[0].roleBriefs.find(
+									(role) =>
+										role.name === 'Partner Sales Users'
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js:157]: move them to a new const file

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js [hunk 6]
```diff
@@ -9,226 +9,214 @@
...
-			sales: 1,
-		},
-	},
-};
-
 export default function () {
-	const [data] = useState(partnerLevelData);
+	const [data, setData] = useState({});
+	const [headcount, setHeadcount] = useState({});
 	const [completed, setCompleted] = useState({});
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js:25]:
```suggestion
	const [data, setData] = useState();
	const [headcount, setHeadcount] = useState();
	const [completed, setCompleted] = useState();
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js [hunk 7]
```diff
@@ -9,226 +9,214 @@
...
+
+								const hastMatchingNPOrNB =
+									accountData.newProjectExistingBusiness ===
+									partnerLevelProperties[
+										accountData.partnerLevel
+									].newProjectExistingBusiness;
+
+								if (hasMatchingARR || hastMatchingNPOrNB) {
+									checkedItems['arr'] = true;
+								}
```
Comment (jairmedeiros) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/src/main/resources/site-initializer/fragments/group/src/dashboard-collection/fragments/partner-level-dashlet/index.js:113]: same here

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

**Comment by JoMaAlves** [comment]
ci:test:relevant

**Comment by thaynam** [comment]
resent [3606](https://github.com/pt-liferay-solutions/liferay-portal/pull/3606)

**Pull request closed without merging**